xla/service/latency_hiding_scheduler.cc

```diff
@@ -1722,9 +1728,20 @@ absl::StatusOr<HloGraphNode::TimeCost> DefaultSchedulerCore::ScheduleNode(
   ++sched_state->scheduled_count;
   for (auto& resource : n->GetResources()) {
     if (resource.second == ResourceUsageType::kResourceRelease) {
-      --sched_state->resources_in_flight[resource.first];
+      sched_state->resource_occupiers_in_flight.at(resource.first)
+          .erase(&n->GetInstr());
     } else if (resource.second == ResourceUsageType::kResourceOccupy) {
-      ++sched_state->resources_in_flight[resource.first];
+      // For async collective done ops, save their corresponding start ops to
+      // the map
+      if (hlo_query::IsAsyncCollectiveDoneOp(&n->GetInstr(), true)) {
+        CHECK(hlo_query::IsAsyncCollectiveStartOp(n->GetInstr().operand(0),
```

Comment:
I don't remember this CHECK-fail from prior version of the code. What's the rationale for it?

Reply:
this check is probably not necessary -- it is making sure we are passing a async collective start op to the following call, while I don't know if there's any case that the operand of a async collective done op is not a start. I think putting it there might help legalize some very weird cases and (hopefully) make the logic easier to read